test/e2e/storage/vsphere_scale.go

```diff
@@ -0,0 +1,144 @@
+		Expect(err).NotTo(HaveOccurred(), "Error Parsing VCP_SCALE_VOLUMES_PER_POD")
+	})
+
+	It("vsphere scale tests", func() {
+		// Volumes will be provisioned with each different types of Storage Class
+		scArrays := make([]*storageV1.StorageClass, 4)
+		// Create default vSphere Storage Class
+		By("Creating Storage Class : sc-default")
+		scDefaultSpec := getVSphereStorageClassSpec("sc-default", nil)
+		scDefault, err := client.StorageV1().StorageClasses().Create(scDefaultSpec)
+		Expect(err).NotTo(HaveOccurred())
+		defer client.StorageV1().StorageClasses().Delete(scDefault.Name, nil)
+		scArrays[0] = scDefault
+
+		// Create Storage Class with vsan storage capabilities
+		By("Creating Storage Class : sc-vsan")
+		var scvsanParameters map[string]string
+		scvsanParameters = make(map[string]string)
+		scvsanParameters[Policy_HostFailuresToTolerate] = "1"
+		scvsanSpec := getVSphereStorageClassSpec("sc-vsan", scvsanParameters)
+		scvsan, err := client.StorageV1().StorageClasses().Create(scvsanSpec)
+		Expect(err).NotTo(HaveOccurred())
+		defer client.StorageV1().StorageClasses().Delete(scvsan.Name, nil)
+		scArrays[1] = scvsan
+
+		// Create Storage Class with SPBM Policy
+		By("Creating Storage Class : sc-spbm")
+		var scSpbmPolicyParameters map[string]string
+		scSpbmPolicyParameters = make(map[string]string)
+		goldPolicy := os.Getenv("VSPHERE_SPBM_GOLD_POLICY")
+		Expect(goldPolicy).NotTo(BeEmpty())
+		scSpbmPolicyParameters[SpbmStoragePolicy] = goldPolicy
+		scSpbmPolicySpec := getVSphereStorageClassSpec("sc-spbm", scSpbmPolicyParameters)
+		scSpbmPolicy, err := client.StorageV1().StorageClasses().Create(scSpbmPolicySpec)
+		Expect(err).NotTo(HaveOccurred())
+		defer client.StorageV1().StorageClasses().Delete(scSpbmPolicy.Name, nil)
+		scArrays[2] = scSpbmPolicy
+
+		// Create Storage Class with User Specified Datastore.
+		By("Creating Storage Class : sc-user-specified-datastore")
+		var scWithDatastoreParameters map[string]string
+		scWithDatastoreParameters = make(map[string]string)
+		datastore := os.Getenv("VSPHERE_DATASTORE")
+		Expect(goldPolicy).NotTo(BeEmpty())
+		scWithDatastoreParameters[Datastore] = datastore
+		scWithDatastoreSpec := getVSphereStorageClassSpec("sc-user-specified-ds", scWithDatastoreParameters)
+		scWithDatastore, err := client.StorageV1().StorageClasses().Create(scWithDatastoreSpec)
+		Expect(err).NotTo(HaveOccurred())
+		defer client.StorageV1().StorageClasses().Delete(scWithDatastore.Name, nil)
+		scArrays[3] = scWithDatastore
+
+		var wg sync.WaitGroup
+		wg.Add(numberOfInstances)
+		volumeCountPerInstance := volumeCount / numberOfInstances
+		for instanceCount := 0; instanceCount < numberOfInstances; instanceCount++ {
+			if instanceCount == numberOfInstances-1 {
+				volumeCountPerInstance = volumeCount
+			}
+			volumeCount = volumeCount - volumeCountPerInstance
+			go CreateVolumesInParallel(client, namespace, scArrays, volumeCountPerInstance, volumesPerPod, &wg)
```

Comment:
Is this function referred from different test?

Reply:
This function CreateVolumesInParallel is renamed to PerformVolumeLifeCycleAtScale at the last minute after your comment on Divyen PR. It is PerformVolumeLifeCycleAtScale instead of CreateVolumesInParallel.